InnerEye/ML/run_ml.py

```diff
@@ -251,75 +270,65 @@ def run(self) -> None:
+
+            if not checkpoint_paths:
+                raise ValueError("Model registration failed: No checkpoints found")
+
+            model_description = "Registering model."
+            checkpoint_paths = checkpoint_paths
             self.register_model_for_epoch(checkpoint_paths, model_description, model_proc)
-            if self.azure_config.register_model_only_for_epoch is not None:
-                return self.azure_config.register_model_only_for_epoch
-
-        # run full image inference on existing or newly trained model on the training, and testing set
-        test_metrics, val_metrics, _ = self.model_inference_train_and_test(checkpoint_handler=checkpoint_handler,
-                                                                           model_proc=model_proc)
-
-        # register the generated model from the run if we haven't already done so
-        if self.model_config.is_segmentation_model and (not self.model_config.is_offline_run):
-            if registration_epoch is None:
-                if self.should_register_model():
-                    assert test_metrics is None or isinstance(test_metrics, InferenceMetricsForSegmentation)
-                    assert val_metrics is None or isinstance(val_metrics, InferenceMetricsForSegmentation)
-                    registration_epoch = self.register_model_for_best_epoch(checkpoint_handler,
-                                                                            test_metrics,
-                                                                            val_metrics,
-                                                                            model_proc)
-            self.try_compare_scores_against_baselines(model_proc)
-        else:
-            logging.warning("Couldn't register model in offline mode")
 
-        return registration_epoch
+        if not self.azure_config.only_register_model:
+            # run full image inference on existing or newly trained model on the training, and testing set
+            test_metrics, val_metrics, _ = self.model_inference_train_and_test(checkpoint_handler=checkpoint_handler,
+                                                                               model_proc=model_proc)
+
+            self.try_compare_scores_against_baselines(model_proc)
 
     def should_register_model(self) -> bool:
         """
         Whether we should register a model at all. If no training has taken place, an equivalent
         model (from the run we recovered) should already have been registered, so we should only
         do so if this run is specifically for that purpose.
         """
-        return self.azure_config.train or self.azure_config.register_model_only_for_epoch is not None
+        return self.azure_config.train or self.azure_config.only_register_model is not None
 
-    def decide_registration_epoch_without_evaluating(self) -> Optional[int]:
+    def decide_registration_epoch_without_evaluating(self,
```

Comment:
This function is redundant, can replace the single call in run_inference_and_register_model with checkpoint_handler.get_checkpoints_to_test()